mutagen/wave.py

```diff
@@ -0,0 +1,369 @@
+            # Normalize ID3v2-tag-chunk to lowercase
+            if chunk.id == u'ID3 ':
+                chunk.id = u'id3 '
+            self.__wavChunks[chunk.id] = chunk
+
+            # Calculate the location of the next chunk,
+            # considering the pad byte
+            self.__next_offset = chunk.offset + chunk.size
+            self.__next_offset += self.__next_offset % 2
+            fileobj.seek(self.__next_offset)
+
+    def __contains__(self, id_):
+        """Check if the RIFF/WAVE file contains a specific chunk"""
+
+        assert_valid_chunk_id(id_)
+
+        return id_ in self.__wavChunks
+
+    def __getitem__(self, id_):
+        """Get a chunk from the RIFF/WAVE file"""
+
+        assert_valid_chunk_id(id_)
+
+        try:
+            return self.__wavChunks[id_]
+        except KeyError:
+            raise KeyError(
+                "%r has no %r chunk" % (self._fileobj, id_))
+
+    def __delitem__(self, id_):
+        """Remove a chunk from the RIFF/WAVE file"""
+
+        assert_valid_chunk_id(id_)
+
+        self.__wavChunks.pop(id_).delete()
+
+    def insert_chunk(self, id_):
+        """Insert a new chunk at the end of the RIFF/WAVE file"""
+
+        assert_valid_chunk_id(id_)
+
+        self._fileobj.seek(self.__next_offset)
+        self._fileobj.write(pack('<4si', id_.ljust(4).encode('ascii'), 0))
+        self._fileobj.seek(self.__next_offset)
+        chunk = RiffChunkHeader(self._fileobj)
+        self[u'RIFF']._update_size(self[u'RIFF'].data_size + chunk.size)
+
+        self.__wavChunks[id_] = chunk
+        self.__next_offset = chunk.offset + chunk.size
+
+
+class WaveStreamInfo(StreamInfo):
+    """WaveStreamInfo()
+
+    Microsoft WAVE file information.
+
+    Information is parsed from the 'fmt ' & 'data'chunk of the RIFF/WAVE file
+
+    Attributes:
+        length (`float`): audio length, in seconds
+        bitrate (`int`): audio bitrate, in bits per second
+        channels (`int`): The number of audio channels
+        sample_rate (`int`): audio sample rate, in Hz
+        sample_size (`int`): The audio sample size
+    """
+
+    length = 0
+    bitrate = 0
+    channels = 0
+    sample_rate = 0
+
+    @convert_error(IOError, error)
+    def __init__(self, fileobj):
+        """Raises error"""
+
+        waveFile = WaveFile(fileobj)
+        try:
+            waveFormatChunk = waveFile[u'fmt ']
+        except KeyError as e:
+            raise error(str(e))
+
+        data = waveFormatChunk.read()
+
+        #  RIFF: http://soundfile.sapp.org/doc/WaveFormat/
+        #  Python struct.unpack:
+        #    https://docs.python.org/2/library/struct.html#byte-order-size-and-alignment
+        info = struct.unpack('<hhLLhh', data[:16])
+        self.audioFormat, self.channels, self.sample_rate, byte_rate, \
+            block_align, self.sample_size = info
+        self.bitrate = self.channels * block_align * self.sample_rate
+
+        try:
+            waveDataChunk = waveFile[u'data']
```

Comment:
It is not checking, it is accessing the data chunk length to calculating the duration (length is seconds) of the audio. I put comment on top of it; removed throwing exception in following line with assigning 0 to number_of_samples.